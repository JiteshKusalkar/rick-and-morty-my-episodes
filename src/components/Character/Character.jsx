import React from 'react';
import styled from 'styled-components';

import { Img, FlexWrapper } from '../../styled';

function Character({ character }) {
  return (
    <Wrapper>
      <Img src={character.image} alt={character.name} />
      <CharacterBio>
        <Name>{character.name}</Name>
        <Field>
          <Label>GENDER: </Label>
          <Gender>{character.gender}</Gender>
        </Field>
        <Field>
          <Label>SPECIES: </Label>
          <Species>{character.species}</Species>
        </Field>
        <Field>
          <Label>STATUS: </Label>
          <Status>{character.status}</Status>
        </Field>
        <Field>
          <Label>TYPE: </Label>
          <Type>{character.type || 'unknown'}</Type>
        </Field>
        <Field>
          <Label>ORIGIN: </Label>
          <Origin>{character.origin.name}</Origin>
        </Field>
        <Field>
          <Label>LOCATION: </Label>
          <Location>{character.location.name}</Location>
        </Field>
      </CharacterBio>
    </Wrapper>
  );
}

export default Character;

const Wrapper = styled(FlexWrapper)`
  margin: 20px 12px;
  width: auto;
`;

const CharacterBio = styled.div`
  width: 100%;
  padding: 15px;

  p {
    margin: 0;
  }
`;

const Field = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 2px;
`;

const Label = styled.label`
  margin-right: 5px;
  font-size: 14px;
  font-weight: 300;
  color: #666666;
`;

const Name = styled.p`
  font-size: 40px;
  font-weight: 300;
  color: #5f9ea0;
`;

const Status = styled.p``;
const Species = styled.p``;
const Type = styled.p``;
const Gender = styled.p``;
const Origin = styled.p``;
const Location = styled.p``;
