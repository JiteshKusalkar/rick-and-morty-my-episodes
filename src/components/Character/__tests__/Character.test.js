import { shallow } from 'enzyme';
import React from 'react';

import Character from '../Character';

// tslint:disable-next-line:no-require-imports
const unroll = require('unroll');
unroll.use(it);

const validProps = {
  character: {
    image: 'https//image-url',
    name: 'Rick',
    gender: 'Male',
    species: 'Human',
    status: 'Alive',
    type: 'Angry Human',
    origin: {
      name: 'Earth'
    },
    location: {
      name: 'Earth'
    }
  }
};

const emptyProps = { character: { ...validProps.character, type: undefined } };

const shallowInstance = props => shallow(<Character {...props} />);

// Snapshot testing
describe('Character', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowInstance(emptyProps);
  });

  unroll(
    'should match the snapshot for #propType',
    (done, args) => {
      wrapper.setProps(args.value);
      expect(wrapper).toMatchSnapshot();
      done();
    },
    [
      ['propType', 'value'],
      ['valid props', validProps],
      ['empty props', emptyProps]
    ]
  );
});
