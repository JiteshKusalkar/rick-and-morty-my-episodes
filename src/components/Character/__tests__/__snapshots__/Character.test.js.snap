// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should match the snapshot for "empty props" 1`] = `
<Styled(styled.div)>
  <styled.img
    alt="Rick"
    src="https//image-url"
  />
  <styled.div>
    <styled.p>
      Rick
    </styled.p>
    <styled.div>
      <styled.label>
        GENDER: 
      </styled.label>
      <styled.p>
        Male
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        SPECIES: 
      </styled.label>
      <styled.p>
        Human
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        STATUS: 
      </styled.label>
      <styled.p>
        Alive
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        TYPE: 
      </styled.label>
      <styled.p>
        unknown
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        ORIGIN: 
      </styled.label>
      <styled.p>
        Earth
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        LOCATION: 
      </styled.label>
      <styled.p>
        Earth
      </styled.p>
    </styled.div>
  </styled.div>
</Styled(styled.div)>
`;

exports[`Character should match the snapshot for "valid props" 1`] = `
<Styled(styled.div)>
  <styled.img
    alt="Rick"
    src="https//image-url"
  />
  <styled.div>
    <styled.p>
      Rick
    </styled.p>
    <styled.div>
      <styled.label>
        GENDER: 
      </styled.label>
      <styled.p>
        Male
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        SPECIES: 
      </styled.label>
      <styled.p>
        Human
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        STATUS: 
      </styled.label>
      <styled.p>
        Alive
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        TYPE: 
      </styled.label>
      <styled.p>
        Angry Human
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        ORIGIN: 
      </styled.label>
      <styled.p>
        Earth
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.label>
        LOCATION: 
      </styled.label>
      <styled.p>
        Earth
      </styled.p>
    </styled.div>
  </styled.div>
</Styled(styled.div)>
`;
